api/core/rag/retrieval/dataset_retrieval.py

```diff
@@ -231,6 +231,13 @@ def retrieve(
                                 source["content"] = segment.content
                             retrieval_resource_list.append(source)
         if hit_callback and retrieval_resource_list:
+            retrieval_resource_list = sorted(
+                retrieval_resource_list, key=lambda x: x.get("score"), reverse=True
+            )
+            position = 1
+            for item in retrieval_resource_list:
+                item["position"] = position
+                position += 1
```

Comment:
```suggestion
            for position, item in enumerate(retrieval_resource_list, start=1):
                item["position"] = position
```

Reply:
And maybe we could generalized a method of tagging position for a list into the `position_helper.py`.